pkg/cmd/cli/secrets/remove_secret_from_object.go

```diff
@@ -0,0 +1,114 @@
+package secrets
+
+import (
+	"fmt"
+	"io"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	RemoveSecretRecommendedName = "remove"
+
+	// TODO: move to examples
+	removeSecretLong = `
+Remove (detach) secrets from a ServiceAccount
+If a secret is no longer valid for a pod, build or image pull, you may remove it from a service account.`
+
+	removeSecretExample = ` // Remove a secret currently associated with a service account like this:
+%[1]s serviceaccount/sa-name secret/secret-name secret/another-secret-name ...`
+)
+
+type RemoveSecretOptions struct {
+	SecretOptions
+}
+
+// NewCmdRemoveSecret creates a command object for detaching one or more secret references from a service account
+func NewCmdRemoveSecret(name, fullName string, f *kcmdutil.Factory, out io.Writer) *cobra.Command {
+	o := &RemoveSecretOptions{SecretOptions{Out: out}}
+
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s serviceaccount/sa-name secret/secret-name [secret/another-secret-name] ...", name),
```

Comment:
I don't think it's clear that `oc secret remove` removes the secret from a service account.  It's possible we chose `add` poorly.  `link` and `unlink` would be appropriate.  However, for consistency we can use `remove`.